nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/data/explorer/PutAzureDataExplorer.java

```diff
@@ -0,0 +1,309 @@
+            .Builder().name("Streaming enabled")
+            .displayName("Streaming enabled")
+            .description("This property determines whether we want to stream data to ADX.")
+            .required(false)
+            .allowableValues("true", "false")
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .defaultValue("false")
+            .build();
+
+    public static final PropertyDescriptor POLL_ON_INGESTION_STATUS = new PropertyDescriptor
+            .Builder().name("Poll on ingestion status")
+            .displayName("Whether to Poll on ingestion status")
+            .description("This property determines whether we want to poll on ingestion status after an ingestion to ADX is completed")
+            .required(false)
+            .allowableValues("true", "false")
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .defaultValue("false")
+            .build();
+    public static final PropertyDescriptor ADX_SERVICE = new PropertyDescriptor
+            .Builder().name("Kusto Ingest Service")
+            .displayName("Kusto Ingest Service")
+            .description("Azure Data Explorer Kusto Ingest Service")
+            .required(true)
+            .identifiesControllerService(KustoIngestService.class)
+            .build();
+
+    static final PropertyDescriptor DATA_FORMAT = new PropertyDescriptor.Builder()
+            .name("Data Format")
+            .displayName("Data Format")
+            .description("The format of the data that is sent to Azure Data Explorer.")
+            .required(true)
+            .allowableValues(KustoIngestDataFormat.values())
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    static final PropertyDescriptor IGNORE_FIRST_RECORD = new PropertyDescriptor.Builder()
+            .name("Ingestion Ignore First Record")
+            .displayName("Ingestion Ignore First Record")
+            .description("Defines whether ignore first record while ingestion.")
+            .required(false)
+            .allowableValues(IGNORE_FIRST_RECORD_YES, IGNORE_FIRST_RECORD_NO)
+            .defaultValue(IGNORE_FIRST_RECORD_NO.getValue())
+            .build();
+
+    static final PropertyDescriptor ROUTE_PARTIALLY_SUCCESSFUL_INGESTION = new PropertyDescriptor.Builder()
+            .name("Route partially successful ingestion records")
+            .displayName("Route partially successful ingestion records")
+            .description("Defines where to route partially successful ingestion records.")
+            .required(false)
+            .allowableValues("Success", "Failure")
+            .defaultValue("Failure")
+            .build();
+
+    static final PropertyDescriptor INGESTION_STATUS_POLLING_TIMEOUT = new PropertyDescriptor.Builder()
+            .name("Timeout for polling on ingestion status")
+            .displayName("Timeout for polling on ingestion status in seconds")
```

Comment:
The name should be shorter and use Title Case.
```suggestion
            .name("Ingest Status Polling Timeout")
            .displayName("Ingest Status Polling Timeout")
```